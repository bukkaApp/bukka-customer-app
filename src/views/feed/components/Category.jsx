import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import shortId from 'shortid';
import InfiniteScroll from 'react-infinite-scroller';
import Row from '../../../components/grid/Row';
import BukkaCard from '../../../components/Carousel/BukkaCard';
import Container from '../../../components/container/Container';
import Navbar from '../../../components/navbar';
import NotAvailable from '../../../components/not-found/NotAvailable';

import fetchBukkaMenuAction from '../../../redux/fetchBukkaMenuAction';
import { useLocationContext } from '../../../context/LocationContext';
import IntroSection from '../common/IntroSection';
import ExploreSection from '../common/ExploreSection';
import getBukkasRelatedToSingleCuisines from '../actionCreators/getBukkasRelatedToSingleCuisines';
import getMoreBukkasRelatedToSingleCuisines from '../actionCreators/getMoreBukkasRelatedToSingleCuisines';


// TODO: Don't  display time if bukkas are not avaailable or they have closed

const PlaceGroup = ({
  name,
  push,
  cuisineItems,
  getBukkasRelatedToCuisines,
  status: { error },
  errorMessage,
  currentPage,
  fetchBukkaMenu,
  getMoreBukkasRelatedToCuisines,
}) => {
  const { coordinates } = useLocationContext();
  const { id } = useParams();
  useEffect(() => {
    let suscribed = true;//eslint-disable-line
    getBukkasRelatedToCuisines(id, coordinates);
    return () => {
      suscribed = false;
    };
  }, [id, coordinates]);

  if (cuisineItems.length === 0 && error) {
    return (
      <div>
        <Navbar push={push} />
        <NotAvailable />
      </div>
    );
  }

  return (
    <div className="Place__Group container-fluid p-0">
      {cuisineItems.length >= 0 && (
        <div>
          <IntroSection push={push} />
          <ExploreSection classNames="pt-5">
            <Container classNames="position-sticky top-114">
              <h2 className="place-group-header pt-100 capitalize pb-3">
                {name}
              </h2>
            </Container>
            <div className="border-top" />
            <Container classNames="position-relative bg-white">
              <InfiniteScroll
                loadMore={() =>
                  getMoreBukkasRelatedToCuisines(
                    id, coordinates, Number(currentPage) + 1)}
                hasMore={errorMessage === ''}
                loader={
                  <div className="loader text-center" key={0}>
                    <div className="spinner-grow custom-text-color" role="status">
                      <span className="sr-only">Loading...</span>
                    </div>
                  </div>
                }
                useWindow
                initialLoad={false}
              >
                <Row classNames="pb-4">
                  {cuisineItems.map(bukka => (
                    <BukkaCard
                      key={shortId.generate()}
                      imageUrl={bukka.imageUrl}
                      mealName={bukka.name}
                      delivery={false}
                      handleClick={() => fetchBukkaMenu(`/bukka/${bukka.slug}`)}
                      deliveryPrice={bukka.deliveryPrice}
                      deliveryTime={bukka.deliveryTime}
                      rating={bukka.rating}
                      tags={bukka.placeGroup}
                      imageHeight="img-fluid"
                      classNames="col-xl-4 col-md-6 col-sm-12"
                      href={`/bukka/${bukka.slug}`}
                    />
                  ))}
                </Row>
              </InfiniteScroll>
            </Container>
          </ExploreSection>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = ({
  deliveryModeReducer: { mode },
  bukkasReducer: { fetchedBukkas, status },
  cuisineReducer: {
    cuisineItems, errorMessage,
    currentPage,
    cuisineToDisplay: { name },
  },
}) => ({
  fetchedBukkas,
  status,
  mode,
  name,
  cuisineItems,
  errorMessage,
  currentPage,
});

export default connect(
  mapStateToProps,
  { getBukkasRelatedToCuisines: getBukkasRelatedToSingleCuisines,
    getMoreBukkasRelatedToCuisines: getMoreBukkasRelatedToSingleCuisines,
    fetchBukkaMenu: fetchBukkaMenuAction, }
)(PlaceGroup);

PlaceGroup.defaultProps = {
  errorMessage: '',
};

PlaceGroup.propTypes = {
  errorMessage: PropTypes.string,
  getMoreBukkasRelatedToCuisines: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  fetchBukkaMenu: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  push: PropTypes.func.isRequired,
  cuisineItems: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  getBukkasRelatedToCuisines: PropTypes.func.isRequired,
  status: PropTypes.objectOf(PropTypes.bool).isRequired
};
