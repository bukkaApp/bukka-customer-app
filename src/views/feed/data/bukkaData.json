[
  {
    "deliveryTime": "40-50 min",
    "title": "burger",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-burger-img.png",
    "rating": "popular",
    "subHeading": "order now",
    "_id": "02994249",
    "position": {
      "top": true
    }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "ice dougnut",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-ice-dougnut.png",
    "rating": "popular",
    "subHeading": "order now",
    "_id": "0234594249",
    "position": {
      "top": true
    }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "alcholic drinks",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-alcholic-drinks.png",
    "rating": "popular",
    "heading": "Get Drinks",
    "_id": "029992034249",
    "subHeading": "ALCOHOL DELIVERED",
    "position": { "top": true }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "fruits",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-fruits.png",
    "rating": "popular",
    "heading": "Get Fresh",
    "_id": "029273561249",
    "subHeading": "Groceries Delivered",
    "position": { "top": true }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "free burger",
    "deliveryPrice": 300,
    "_id": "0299425723234",
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-free-burger.png",
    "rating": "popular",
    "heading": "Get All Orders Delivered Free",
    "subHeading": "TRY UNLIMITED",
    "textPositionBottom": true
  }
]
