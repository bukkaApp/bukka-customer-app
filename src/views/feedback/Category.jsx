import React from 'react';
// import PropTypes from 'prop-types';
import Footer from '../../components/footer/Footer';
import BannerSection from '../support/common/BannerSection';
import PersonalizedHeader from '../support/components/PersonalizedHeader';
import PersonalizedBody from '../support/components/PersonalizedBody';
import postData from '../support/inputData/complains.json';

import './category.scss';
import '../support/components/supportmainsection.scss';

const Category = () => (
  <div>
    <BannerSection />
    <div className="support-main-section-container">
      <div className="col-md-12">
        <PersonalizedHeader
          className="unselectable category_heading__text"
          title="What can we help you with?"
        />
        {postData.category.map(topic => (
          <PersonalizedBody text={topic.text} link={topic.link} />
        ))}
      </div>
    </div>
    <Footer />
  </div>

);

export default Category;

Category.propTypes = {
  // authenticated: PropTypes.bool.isRequired,
};

