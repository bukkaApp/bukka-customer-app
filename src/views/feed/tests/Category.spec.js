import React from 'react';

import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import freeDelivery from '../data/free-delivery.json';
import bukkaData from '../data/search.json';
import Category from 'src/components/Category';

const initialState = {
  navbarAuthReducer: { type: 'Sign In', },
  deliveryModeReducer: { mode: 'delivery', },
  locationsPredictionReducer: { predictions: [3.7474, 3.4848] },
  authenticationReducer: {
    status: {
      authenticated: false,
      error: false
    }
  },
  cuisineReducer: {
    cuisineItems: [{
      _id: '5ddc5dfb9a29af135cd21ce7',
      name: 'american',
      imageUrl: 'https://res.cloudinary.com/mybukka/image/upload/v1574723013/American_e1n3aj.png'
    }],
    errorMessage: '',
    currentPage: 1,
    cuisineToDisplay: { name: 'cuisine name' },
  },
  bukkasReducer: {
    fetchedBukkas: {
      nearbyBukkas: [],
      message: ''
    },
    status: {
      fetchedBukkas: false,
      error: true,
    }
  },
  selectedLocationReducer: { coordinates: [3.7474, 3.4848] }
};

const store = mockStore(initialState);

describe('Category component', () => {
  const props = {
    signOut: jest.fn(),
    match: {
      params: { id: 'ameriacan' },
    },
    setDeliveryModeAction: jest.fn(),
    push: jest.fn(),
    coordinates: [3.7474, 3.4848],
    fetchedBukkas: { nearbyBukkas: [] },
    fetchNearbyBukkas: jest.fn(),
    status: { error: false }
  };
  const { container, rerender } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Category {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('renders properly', () => {
    expect(container).toMatchSnapshot();
  });
  it('renders properly', () => {
    const nearbyBukkas = [...freeDelivery, ...bukkaData];
    const newState = {
      ...initialState,
      bukkasReducer: {
        ...initialState.bukkasReducer,
        fetchedBukkas: {
          ...initialState.bukkasReducer.fetchedBukkas,
          nearbyBukkas,
          message: ''
        },
        status: {
          ...initialState.bukkasReducer.status,
          fetchedBukkas: true,
          error: false
        }
      }
    };
    const newProps = {
      ...props,
      fetchedBukkas: { nearbyBukkas },
    };
    const newStore = mockStore(newState);
    rerender(
      <Provider store={newStore}>
        <MemoryRouter>
          <Category {...newProps} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
