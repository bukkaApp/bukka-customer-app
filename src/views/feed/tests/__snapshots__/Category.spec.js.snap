// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component renders properly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        aria-hidden="true"
        aria-labelledby="exampleModalLabel"
        class="modal fade"
        id="authModal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered auth-modal"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            />
            <div
              class="modal-body"
            >
              <div
                class="dismiss-modal close"
                data-dismiss="modal"
              >
                <svg
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <defs>
                    <path
                      d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                      id="icon-close_svg__a"
                    />
                  </defs>
                  <use
                    transform="translate(4 4)"
                    xlink:href="#icon-close_svg__a"
                  />
                </svg>
              </div>
              <div
                class="bg-color auth-page"
              >
                <div
                  class=""
                >
                  <div
                    class="auth-wrapper pt-5"
                  >
                    <h2
                      class="text-center pt-4 mb-0 auth-title"
                    >
                      Log In
                    </h2>
                    <form
                      class="pb-3 form-auth"
                      name="form"
                    >
                      <div
                        class="form-title mb-4"
                      >
                        <span>
                          Type your email
                        </span>
                      </div>
                      <div
                        class="input-group-wrapper mt-9"
                      >
                        <div
                          class="input-slide text-center "
                        >
                          <div
                            class="form-group padding"
                          >
                            <input
                              accept=""
                              autocomplete="off"
                              class="default-input"
                              name="email"
                              placeholder="Email"
                              type="email"
                              value=""
                            />
                            <div
                              class="text-danger help-block"
                            />
                          </div>
                          <div
                            class="form-group padding"
                          >
                            <input
                              accept=""
                              autocomplete="off"
                              class="default-input"
                              name="password"
                              placeholder="Password"
                              type="password"
                              value=""
                            />
                            <div
                              class="text-danger help-block"
                            />
                          </div>
                        </div>
                        <div
                          aria-pressed="false"
                          class="col-lg-12 padding terms text-center"
                          role="button"
                          tabindex="0"
                        >
                          <p
                            class="link text-muted"
                          >
                            Forgot Password?
                          </p>
                        </div>
                      </div>
                      <div
                        class="padding"
                      >
                        <button
                          class="button col-md-12 primary-button mt-5"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="submit"
                          type="submit"
                        >
                          Log In
                        </button>
                      </div>
                      <div
                        class="divider-box mt-2 mb-2"
                      >
                        <div
                          class="divider"
                        >
                          <span>
                            or
                          </span>
                        </div>
                      </div>
                      <div
                        class="padding"
                      >
                        <button
                          class="facebk-btn col-md-12"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="aY-tiu0KU"
                          type="button"
                        >
                          Facebook
                        </button>
                      </div>
                      <div
                        class="form-options padding"
                      >
                        <p>
                          New to Bukka?
                        </p>
                        <button
                          class="btn-link auth-footer-btn"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="/signup"
                          type="button"
                        >
                          SIGN UP
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class=""
    >
      <nav
        class="container navbar navbar-light"
      >
        <div>
          <a
            class="bukka-brand"
            href="/"
          >
            <div
              class="bukka-logo"
            >
              <img
                alt=""
                src="https://res.cloudinary.com/digitzs/image/upload/v1550480450/Bukka%20App/bukka-logo.svg"
              />
            </div>
          </a>
        </div>
      </nav>
    </div>
    <div
      class="bg-color not-available-page"
    >
      <div
        class="container "
      >
        <h1
          class="text-title-unavailable"
        >
          Mybuka Not
          <br />
          Available In Your
          <br />
          Location
          <br />
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category component renders properly 2`] = `
<div>
  <div
    class="container-fluid p-0"
  >
    <div>
      <div
        class="intro-navbar-area"
      >
        <div
          class="container"
        >
          <div
            aria-hidden="true"
            aria-labelledby="exampleModalLabel"
            class="modal fade"
            id="authModal"
            role="dialog"
            tabindex="-1"
          >
            <div
              class="modal-dialog modal-dialog-centered auth-modal"
              role="document"
            >
              <div
                class="modal-content"
              >
                <div
                  class="modal-header"
                />
                <div
                  class="modal-body"
                >
                  <div
                    class="dismiss-modal close"
                    data-dismiss="modal"
                  >
                    <svg
                      fill="currentColor"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <defs>
                        <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />
                      </defs>
                      <use
                        transform="translate(4 4)"
                        xlink:href="#icon-close_svg__a"
                      />
                    </svg>
                  </div>
                  <div
                    class="bg-color auth-page"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="auth-wrapper pt-5"
                      >
                        <h2
                          class="text-center pt-4 mb-0 auth-title"
                        >
                          Log In
                        </h2>
                        <form
                          class="pb-3 form-auth"
                          name="form"
                        >
                          <div
                            class="form-title mb-4"
                          >
                            <span>
                              Type your email
                            </span>
                          </div>
                          <div
                            class="input-group-wrapper mt-9"
                          >
                            <div
                              class="input-slide text-center "
                            >
                              <div
                                class="form-group padding"
                              >
                                <input
                                  accept=""
                                  autocomplete="off"
                                  class="default-input"
                                  name="email"
                                  placeholder="Email"
                                  type="email"
                                  value=""
                                />
                                <div
                                  class="text-danger help-block"
                                />
                              </div>
                              <div
                                class="form-group padding"
                              >
                                <input
                                  accept=""
                                  autocomplete="off"
                                  class="default-input"
                                  name="password"
                                  placeholder="Password"
                                  type="password"
                                  value=""
                                />
                                <div
                                  class="text-danger help-block"
                                />
                              </div>
                            </div>
                            <div
                              aria-pressed="false"
                              class="col-lg-12 padding terms text-center"
                              role="button"
                              tabindex="0"
                            >
                              <p
                                class="link text-muted"
                              >
                                Forgot Password?
                              </p>
                            </div>
                          </div>
                          <div
                            class="padding"
                          >
                            <button
                              class="button col-md-12 primary-button mt-5"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="submit"
                              type="submit"
                            >
                              Log In
                            </button>
                          </div>
                          <div
                            class="divider-box mt-2 mb-2"
                          >
                            <div
                              class="divider"
                            >
                              <span>
                                or
                              </span>
                            </div>
                          </div>
                          <div
                            class="padding"
                          >
                            <button
                              class="facebk-btn col-md-12"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="aY-tiu0KU"
                              type="button"
                            >
                              Facebook
                            </button>
                          </div>
                          <div
                            class="form-options padding"
                          >
                            <p>
                              New to Bukka?
                            </p>
                            <button
                              class="btn-link auth-footer-btn"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="/signup"
                              type="button"
                            >
                              SIGN UP
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <nav
          class="container navbar navbar-light"
        >
          <div
            class="row mx-0"
          >
            <div>
              <a
                class="bukka-brand"
                href="/"
              >
                <div
                  class="bukka-logo"
                >
                  <img
                    alt=""
                    src="https://res.cloudinary.com/digitzs/image/upload/v1550480450/Bukka%20App/bukka-logo.svg"
                  />
                </div>
              </a>
            </div>
            <div
              class="pl-lg-5
            d-none d-md-none d-lg-inline-flex"
            >
              <div
                class="pr-17"
              >
                <div
                  class="position-relative"
                >
                  <div>
                    <button
                      class="current-location-button inherit-bg"
                      data-dismiss=""
                      data-target=""
                      data-testid="button"
                      data-toggle=""
                      id="aY-tiu0KU"
                      type="button"
                    >
                      <span
                        class="current-location-button-icon custom-mt-minus19"
                      >
                        <svg
                          class="magnifier"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <defs>
                            <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />
                          </defs>
                          <use
                            fill="currentColor"
                            fill-rule="evenodd"
                            opacity=".9"
                            xlink:href="#icon-search_svg__a"
                          />
                        </svg>
                      </span>
                      <div>
                        <div
                          class="current-location-button-text"
                        >
                          <input
                            accept=""
                            autocomplete="searchanything"
                            class="searchanything-text-field inherit-bg searchlocation"
                            name="searchLocation"
                            placeholder="Search for anything..."
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-inline"
          >
            <div
              class="d-none bukka-md-inline-flex"
            >
              <a
                class="bukka-btn"
                href="/feed"
              >
                Food
              </a>
              <a
                class="bukka-btn"
                href="/fresh"
              >
                Fresh
              </a>
              <a
                class="bukka-btn"
                href="/drinks"
              >
                Drinks
              </a>
            </div>
            <button
              class="small-outline-button bg-transparent"
              data-dismiss=""
              data-target="#authModal"
              data-testid="button"
              data-toggle="modal"
              id="/login"
              type="button"
            >
              sign in
            </button>
            <button
              class="small-button mr-0"
              data-dismiss=""
              data-target="#authModal"
              data-testid="button"
              data-toggle="modal"
              id="/signup"
              type="button"
            >
              sign up
            </button>
          </div>
        </nav>
      </div>
      <section
        class="feed-main-section pt-5"
      >
        <div
          class="container "
        >
          <h2
            class="pt-100 px-15 capitalize"
          >
            Fast food
          </h2>
        </div>
        <div
          class="border-top"
        />
        <div
          class="container "
        >
          <div
            class="mt-4 mb-4"
          >
            hehehee
            
            
            <div
              class="container "
            >
              <div
                class="row pb-4"
              >
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-7.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-7.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            miss saigon
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595379/bukka-favorite-5.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595379/bukka-favorite-5.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            popsons
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595379/bukka-favorite-6.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595379/bukka-favorite-6.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            the new spot on pork
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-8.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-8.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            chaat corner
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595379/bukka-favorite-1.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595379/bukka-favorite-1.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            fresh roll
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595379/bukka-favorite-2.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595379/bukka-favorite-2.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            jack in the box
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-3.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-3.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            ike's love and sandwishes
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container col-lg-4 col-md-4 col-sm-12"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target="#mealModal"
                      data-toggle="modal"
                      role="button"
                      tabindex="0"
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-4.webp"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-4.webp"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            Urban pizza kitchen
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
